tests/generic_views/test_edit.py

```diff
@@ -214,34 +214,37 @@ class MyCreateView(CreateView):
 @override_settings(ROOT_URLCONF='generic_views.urls')
 class UpdateViewTests(TestCase):
 
-    def test_update_post(self):
-        a = Author.objects.create(
+    @classmethod
+    def setUpTestData(cls):
+        cls.author = Author.objects.create(
+            pk=1,
             name='Randall Munroe',
             slug='randall-munroe',
         )
-        res = self.client.get('/edit/author/%d/update/' % a.pk)
+
+    def test_update_post(self):
+        res = self.client.get('/edit/author/%d/update/' % self.author.pk)
         self.assertEqual(res.status_code, 200)
         self.assertIsInstance(res.context['form'], forms.ModelForm)
-        self.assertEqual(res.context['object'], Author.objects.get(pk=a.pk))
-        self.assertEqual(res.context['author'], Author.objects.get(pk=a.pk))
+        self.assertEqual(res.context['object'], Author.objects.get(pk=self.author.pk))
```

Comment:
I don't see the need to refetch the object from the database.
`self.assertEqual(res.context['object'], self.author)`
should work fine for all these assertions. Maybe the original test author didn't realize that model equality only compares primary keys.